pkg/controller/core.oam.dev/v1alpha2/application/application_controller.go

```diff
@@ -247,22 +249,42 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 
 	app.Status.Phase = common.ApplicationHealthChecking
 	klog.Info("Check application health status")
-	// check application health status
-	appCompStatus, healthy, err := handler.aggregateHealthStatus(appFile)
-	if err != nil {
-		klog.ErrorS(err, "Failed to aggregate status", "application", klog.KObj(app))
-		r.Recorder.Event(app, event.Warning(velatypes.ReasonFailedHealthCheck, err))
-		return r.endWithNegativeCondition(ctx, app, condition.ErrorCondition("HealthCheck", err))
-	}
-	app.Status.Services = appCompStatus
-	if !healthy {
-		if err := r.patchStatus(ctx, app); err != nil {
-			return r.endWithNegativeCondition(ctx, app, condition.ReconcileError(err))
+
+	// check application health status if no health check policy is applied
+	if !hasHealthCheckPolicy(appFile.Policies) {
+		appCompStatus, healthy, err := handler.aggregateHealthStatus(appFile)
+		if err != nil {
+			klog.ErrorS(err, "Failed to aggregate status", "application", klog.KObj(app))
```

Comment:
The line of this log might be too large. Logging the application's namespace and name is enough.

Reply:
sure